src/App.tsx

```diff
@@ -15,37 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SetByReverse = true;
+
+enum SortedFields {
+  SetById,
+  SetByAlphabet,
+  SetByLength,
+  SetByDanger,
+}
+
+function getSortedGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortedFields; reversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortedFields.SetById:
+      case SortedFields.SetByAlphabet:
+        return good1.localeCompare(good2);
+      case SortedFields.SetByLength:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setsortField] = useState(SortedFields.SetByDanger);
+  const [reversed, setReversed] = useState(false);
+
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortField, reversed });
+
+  const onReset = () => {
+    setReversed(false);
+    setsortField(SortedFields.SetByDanger);
+  };
+
+  const onReverse = () => {
+    setReversed(!reversed);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setsortField(SortedFields.SetByAlphabet)}
+          type="button"
+          className={cn(
+            'button is-info',
+
+            { 'is-light': sortField !== SortedFields.SetByAlphabet },
+          )}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setsortField(SortedFields.SetByLength)}
+          type="button"
+          className={cn(
+            'button is-success',
+
+            { 'is-light': sortField !== SortedFields.SetByLength },
+          )}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={onReverse}
+          type="button"
+          className={cn(
+            'button is-warning',
+
+            { 'is-light': reversed !== SetByReverse },
+          )}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(reversed || sortField !== SortedFields.SetByDanger) && (
```

Comment:
The condition for rendering the 'Reset' button is checking if 'sortField' is not 'SetByDanger', but since 'SetByDanger' does not have an associated sorting logic, this condition might not work as expected. Ensure that the condition correctly reflects when the 'Reset' button should be displayed.